docs/src/theme/NavbarItem/DocsVersionDropdownNavbarItem.tsx

```diff
@@ -3,39 +3,145 @@
  * Licensed under the MIT License.
  */
 
-import { useActiveDocContext } from "@docusaurus/plugin-content-docs/client";
-import type { WrapperProps } from "@docusaurus/types";
-import type DocsVersionDropdownNavbarItemType from "@theme/NavbarItem/DocsVersionDropdownNavbarItem";
-import DocsVersionDropdownNavbarItem from "@theme-original/NavbarItem/DocsVersionDropdownNavbarItem";
+import { translate } from "@docusaurus/Translate";
+import {
+	useVersions,
+	useActiveDocContext,
+	useDocsVersionCandidates,
+	useDocsPreferredVersion,
+} from "@docusaurus/plugin-content-docs/client";
+import type {
+	GlobalVersion,
+	GlobalDoc,
+	ActiveDocContext,
+} from "@docusaurus/plugin-content-docs/client";
+import { useLocation } from "@docusaurus/router";
+import type { LinkLikeNavbarItemProps } from "@theme/NavbarItem";
+import DefaultNavbarItem from "@theme/NavbarItem/DefaultNavbarItem";
+import type { Props } from "@theme/NavbarItem/DocsVersionDropdownNavbarItem";
+import DropdownNavbarItem from "@theme/NavbarItem/DropdownNavbarItem";
 import React from "react";
 
 /**
- * Note: this module was generated by `swizzling` the Docusaurus "classic" theme's Footer component.
- *
- * Its inner layout and contents were updated to better suite our needs, but the underlying styling and inner
- * components remain unchanged.
+ * Gets the documentation page marked as the main/landing page for a version,
+ * identified by the mainDocId property.
  *
- * See {@link https://docusaurus.io/docs/swizzling/ | here} for more information on swizzling.
+ * @param version - The version object to get the main doc from
+ * @returns The main documentation page for this version
  */
+function getVersionMainDoc(version: GlobalVersion): GlobalDoc {
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	return version.docs.find((doc) => doc.id === version.mainDocId)!;
```

Comment:
The main doc should always exist right? Should we considering adding an assertion here?

Reply:
This is a good point. We should explicitly mention the expected behavior in edge cases, such as when no main doc is found

Reply:
What method of assertion would you prefer? I used the non-null assertion, we could use another method. should I assign it to a variable and assert if that variable is not unefined?